app/controllers/post_controller.rb

```diff
@@ -1,9 +1,40 @@
 class PostController < ApplicationController
   def index
-    @post = User.find(params[:user_id]).post
+    user_id = params[:user_id]
+    @user = User.find(user_id)
+    @posts = @user.posts.order(created_at: :desc)
   end
 
   def show
-    @post = User.find(params[:user_id]).posts.find(params[:id])
+    user_id = params[:user_id]
+    id = params[:id]
+    @user = User.find(user_id)
+    @post = Post.find(id)
+    @comments = @post.comments.order(created_at: :desc)
+    @likes = @post.likes.all
+  end
+
+  def new
+    @current = current_user
+  end
```

Comment:
- [ ] the new def should be for the post but the one you are adding should be placed in the `User_controller.rb`.

Reply:
Done